src/receive_imf.rs

```diff
@@ -546,19 +546,27 @@ async fn add_parts(
         // signals whether the current user is a bot
         let is_bot = context.get_config_bool(Config::Bot).await?;
 
-        let create_blocked = match test_normal_chat {
-            Some(ChatIdBlocked {
-                id: _,
-                blocked: Blocked::Request,
-            }) if is_bot => Blocked::Not,
-            Some(ChatIdBlocked { id: _, blocked }) => blocked,
-            None => {
-                if is_bot {
-                    Blocked::Not
-                } else {
-                    Blocked::Request
+        let create_blocked = if let Some(ChatIdBlocked { id: _, blocked }) = test_normal_chat {
+            match blocked {
+                Blocked::Request => {
+                    if is_bot {
+                        Blocked::Not
+                    } else {
+                        Blocked::Request
+                    }
+                }
+                Blocked::Not | Blocked::Yes => {
+                    if Contact::is_blocked_load(context, from_id).await? {
```

Comment:
It's unfortunate that we need to lookup the contact from the db once again here. It makes it worth even more to load the contact struct once in the beginning of receive_imf, and then pass that around instead of `from_id`.